Content.Shared/_RMC14/Xenonids/Acid/SharedXenoAcidSystem.cs

```diff
@@ -99,16 +118,27 @@ public override void Update(float frameTime)
         if (_net.IsClient)
             return;
 
-        var query = EntityQueryEnumerator<CorrodingComponent>();
         var time = _timing.CurTime;
 
-        while (query.MoveNext(out var uid, out var corroding))
+        var damageableCorrodingQuery = EntityQueryEnumerator<DamageableCorrodingComponent>();
+        while (damageableCorrodingQuery.MoveNext(out var uid, out var damageableCorrodingComponent))
+        {
+            if (time > damageableCorrodingComponent.LastDamagedAt.Add(TimeSpan.FromSeconds(_corrosiveAcidTickDelaySeconds)))
```

Comment:
You can store this add as a  NextDamageAt field on damageable corroding component, then you don't need to add it every time this is checked

Reply:
Sure. Done.